pkg/generic/generic_service.go

```diff
@@ -35,21 +36,39 @@ type Service interface {
 }
 
 // ServiceInfo create a generic ServiceInfo
-func ServiceInfo(pcType serviceinfo.PayloadCodec) *serviceinfo.ServiceInfo {
-	return newServiceInfo(pcType)
+func ServiceInfo(pcType serviceinfo.PayloadCodec, codec serviceinfo.CodecInfo) *serviceinfo.ServiceInfo {
```

Comment:
Shouldn't we change existing ServiceInfo?
Better create a new api to create a ServiceInfo with codec?

Reply:
Since all fields of ServiceInfo are public, you can new it and then set the codec